libvast/src/system/catalog.cpp

```diff
@@ -396,54 +410,119 @@ catalog_state::lookup_impl(const expression& expr) const {
 
 catalog_actor::behavior_type
 catalog(catalog_actor::stateful_pointer<catalog_state> self,
-        accountant_actor accountant) {
+        accountant_actor accountant, const std::filesystem::path& dir) {
   if (self->getf(caf::local_actor::is_detached_flag))
     caf::detail::set_thread_name("vast.catalog");
   self->state.self = self;
+  self->state.type_registry_dir = dir;
```

Comment:
Either the argument name or the state member variable name here are very misleading.

Reply:
Elaborated in https://github.com/tenzir/vast/pull/2745/commits/af7e9288510ad1263be60c40be7d35a71d02dbdd